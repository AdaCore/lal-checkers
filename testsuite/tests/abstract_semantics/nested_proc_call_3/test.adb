procedure Test is
   procedure F;

   x : Integer := 0;
   y : Integer := 42;

   procedure G is
   begin
      F;
      y := 12;
      F;
   end G;

   procedure F is
   begin
      x := x + 1;
   end F;
begin
   F;
   F;
   G;
end Ex1;
