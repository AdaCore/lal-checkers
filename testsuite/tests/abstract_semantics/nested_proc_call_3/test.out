{
  "always": {
    "assign0": [
      {
        "trace:": [
          "assign0", 
          "start0"
        ],
        "values": {
          "$stack": "({1: '[0, 0]'}, 0)"
        }
      }
    ],
    "assign1": [
      {
        "trace:": [
          "assign0",
          "assign1",
          "start0"
        ],
        "values": {
          "$stack": "({1: '[0, 0]', 2: '[42, 42]'}, 0)"
        }
      }
    ],
    "assign2": [
      {
        "trace:": [
          "assign0",
          "assign1",
          "assign2",
          "start0"
        ],
        "values": {
          "$stack": "({1: '[0, 0]', 2: '[42, 42]'}, 0)"
        }
      }
    ],
    "assign3": [
      {
        "trace:": [
          "assign0",
          "assign1",
          "assign2",
          "assign3",
          "start0"
        ],
        "values": {
          "$stack": "({1: '[1, 1]', 2: '[42, 42]'}, 0)"
        }
      }
    ],
    "assign4": [
      {
        "trace:": [
          "assign0",
          "assign1",
          "assign2",
          "assign3",
          "assign4",
          "start0"
        ],
        "values": {
          "$stack": "({1: '[1, 1]', 2: '[42, 42]'}, 0)"
        }
      }
    ],
    "assign5": [
      {
        "trace:": [
          "assign0",
          "assign1",
          "assign2",
          "assign3",
          "assign4",
          "assign5",
          "start0"
        ],
        "values": {
          "$stack": "({1: '[2, 2]', 2: '[42, 42]'}, 0)"
        }
      }
    ],
    "assign6": [
      {
        "trace:": [
          "assign0",
          "assign1",
          "assign2",
          "assign3",
          "assign4",
          "assign5",
          "assign6",
          "start0"
        ],
        "values": {
          "$stack": "({1: '[2, 2]', 2: '[42, 42]'}, 0)"
        }
      }
    ],
    "assign7": [
      {
        "trace:": [
          "assign0",
          "assign1",
          "assign2",
          "assign3",
          "assign4",
          "assign5",
          "assign6",
          "assign7",
          "start0"
        ],
        "values": {
          "$stack": "({1: '[4, 4]', 2: '[12, 12]'}, 0)"
        }
      }
    ],
    "start0": [
      {
        "trace:": [
          "start0"
        ],
        "values": {
          "$stack": "({}, 0)"
        }
      }
    ]
  },
  "le_t_eq_v": {
    "assign0": [
      {
        "trace:": [
          "assign0",
          "start0"
        ],
        "values": {
          "$stack": "({1: '[0, 0]'}, 0)"
        }
      }
    ],
    "assign1": [
      {
        "trace:": [
          "assign0",
          "assign1",
          "start0"
        ],
        "values": {
          "$stack": "({1: '[0, 0]', 2: '[42, 42]'}, 0)"
        }
      }
    ],
    "assign2": [
      {
        "trace:": [
          "assign0",
          "assign1",
          "assign2",
          "start0"
        ],
        "values": {
          "$stack": "({1: '[0, 0]', 2: '[42, 42]'}, 0)"
        }
      }
    ],
    "assign3": [
      {
        "trace:": [
          "assign0",
          "assign1",
          "assign2",
          "assign3",
          "start0"
        ],
        "values": {
          "$stack": "({1: '[1, 1]', 2: '[42, 42]'}, 0)"
        }
      }
    ],
    "assign4": [
      {
        "trace:": [
          "assign0",
          "assign1",
          "assign2",
          "assign3",
          "assign4",
          "start0"
        ],
        "values": {
          "$stack": "({1: '[1, 1]', 2: '[42, 42]'}, 0)"
        }
      }
    ],
    "assign5": [
      {
        "trace:": [
          "assign0",
          "assign1",
          "assign2",
          "assign3",
          "assign4",
          "assign5",
          "start0"
        ],
        "values": {
          "$stack": "({1: '[2, 2]', 2: '[42, 42]'}, 0)"
        }
      }
    ],
    "assign6": [
      {
        "trace:": [
          "assign0",
          "assign1",
          "assign2",
          "assign3",
          "assign4",
          "assign5",
          "assign6",
          "start0"
        ],
        "values": {
          "$stack": "({1: '[2, 2]', 2: '[42, 42]'}, 0)"
        }
      }
    ],
    "assign7": [
      {
        "trace:": [
          "assign0",
          "assign1",
          "assign2",
          "assign3",
          "assign4",
          "assign5",
          "assign6",
          "assign7",
          "start0"
        ],
        "values": {
          "$stack": "({1: '[4, 4]', 2: '[12, 12]'}, 0)"
        }
      }
    ],
    "start0": [
      {
        "trace:": [
          "start0"
        ],
        "values": {
          "$stack": "({}, 0)"
        }
      }
    ]
  }
}